compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -2775,7 +2775,11 @@ extends SyntaxMsg(TargetNameOnTopLevelClassID):
 class NotClassType(tp: Type)(using Context)
 extends TypeMsg(NotClassTypeID), ShowMatchTrace(tp):
   def msg(using Context) = i"$tp is not a class type"
-  def explain(using Context) = ""
+  def explain(using Context) =
+    i"""A class type includes classes and traits in a specific order. Defining a class, even an anonymous class,
+        |requires specifying a linearization order for the traits it extends. For example, `A & B` is not a class type
+        |because it doesn't specify which trait takes precedence, A or B. Both the list `A,B` and `B,A` are class types.
+        |Class types also can't have refinements."""
```

Comment:
I like the suggestion on the ticket to add:
```
Did you mean `Serializable, Product`?
```
or
```
Did you mean `Serializable with Product`?
```
as context dictates. I would offer that without `-explain`. Another view of messaging that comes up in this repo is that the error is a prompt for correction, and usually the user either knows right away what to do or needs a quick reminder only.

I agree that if the explain does not have diagnostic info, then some examples for the "gist" of it is most helpful, rather than an exhaustive definition of terms.

I don't know if linking to the spec is more explanatory than the general docs, though the URL will be more stable.

`Ask your team lead what they think a class type is.`

Reply:
> I don't know if linking to the spec is more explanatory than the general docs

Is there a general docs page that talks about linearization?

Reply:
Because we have multiple contexts in which this error could be thrown, it would only confuse users more to see explanations like `class A extends Object, (Serializable & Product)` if this error is being thrown elsewhere. Because there's no easy fix, I suggest providing a simple example without a suggestion that may not work depending on the context (e.g. `A with B` may not work everywhere). I will make an issue in the documentation repo to add a page about class types so that users don't have to read the language spec, but in the meantime add a reference to the specification in the error message so users can read up about class types.